src/SourceGenerators/Uno.UI.Tasks/RuntimeAssetsSelector/RuntimeAssetsSelectorTask.cs

```diff
@@ -0,0 +1,119 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Runtime.InteropServices.ComTypes;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Transactions;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+
+namespace Uno.UI.Tasks.RuntimeAssetsSelector
+{
+	/// <summary>
+	/// A task used to merge linker definition files and embed the result in an assembly
+	/// </summary>
+	public class RuntimeAssetsSelectorTask_v0 : Microsoft.Build.Utilities.Task
+	{
+		[Required]
+		public Microsoft.Build.Framework.ITaskItem[]? UnoRuntimeEnabledPackage { get; set; }
+
+		[Required]
+		public string UnoRuntimeIdentifier { get; set; } = "";
+
+		/// <remarks>
+		/// Note that this property is not set to [Required] because
+		/// with netstandard2.0, it is not set and the default value is used instead.
+		/// </remarks>
+		public string TargetFrameworkVersion { get; set; } = "";
+
+		[Output]
+		public Microsoft.Build.Framework.ITaskItem[]? Assemblies { get; set; }
+
+		[Output]
+		public Microsoft.Build.Framework.ITaskItem[]? DebugSymbols { get; set; }
+
+		public override bool Execute()
+		{
+			try
+			{
+				List<ITaskItem> assemblies = new();
+				List<ITaskItem> debugSymbols = new();
+
+				foreach (var package in UnoRuntimeEnabledPackage ?? Array.Empty<ITaskItem>())
+				{
+					var packageBasePath = package.GetMetadata("PackageBasePath");
+
+					if (!Version.TryParse(TargetFrameworkVersion?.Substring(1), out var targetFrameworkVersion))
```

Comment:
@jeromelaban  if `TargetFrameworkVersion` is not set, meaning its value is `""` (string.Empty)  would not `Substring(1)` result in a `System.ArgumentOutOfRangeException`?

```
    "".Substring(1);
```

Reply:
Indeed, and thinking about it, since we're not allowing netstandard2.0 anymore for those builds (net7.0+) TargetFramework will never be null after all.